pkg/peertls/tlsopts/options.go

```diff
@@ -64,30 +71,58 @@ func (opts *Options) configure(c Config) (err error) {
 			return Error.Wrap(err)
 		}
 		parseOpts.CAWhitelist = parsed
-		pcvs = append(pcvs, peertls.VerifyCAWhitelist(parsed))
+		opts.VerificationFuncs.ClientAdd(peertls.VerifyCAWhitelist(parsed))
 	}
 
 	if c.Extensions.Revocation {
 		opts.RevDB, err = identity.NewRevDB(c.RevocationDBURL)
 		if err != nil {
 			return err
 		}
-		pcvs = append(pcvs, identity.VerifyUnrevokedChainFunc(opts.RevDB))
+		opts.VerificationFuncs.Add(identity.VerifyUnrevokedChainFunc(opts.RevDB))
 	}
 
 	exts := peertls.ParseExtensions(c.Extensions, parseOpts)
-	pcvs = append(pcvs, exts.VerifyFunc())
+	opts.VerificationFuncs.Add(exts.VerifyFunc())
+
+	opts.Cert, err = peertls.TLSCert(opts.Ident.RawChain(), opts.Ident.Leaf, opts.Ident.Key)
+	return err
+}
+
+// Client returns the client verification functions.
+func (vf *VerificationFuncs) Client() []peertls.PeerCertVerificationFunc {
+	return vf.client
+}
+
+// Server returns the server verification functions.
+func (vf *VerificationFuncs) Server() []peertls.PeerCertVerificationFunc {
+	return vf.server
+}
 
-	// NB: remove nil elements
-	for i, f := range pcvs {
+// Add adds verification functions so the client and server lists.
+func (vf *VerificationFuncs) Add(verificationFuncs ...peertls.PeerCertVerificationFunc) {
+	vf.ClientAdd(verificationFuncs...)
+	vf.ServerAdd(verificationFuncs...)
+}
+
+// ClientAdd adds verification functions so the client list.
+func (vf *VerificationFuncs) ClientAdd(verificationFuncs ...peertls.PeerCertVerificationFunc) {
+	verificationFuncs = removeNils(verificationFuncs)
```

Comment:
why do the nil funcs need to be removed? when would they usually be set to nil?

Reply:
some of the factories for verification functions like the one for peer CA whitelist verification return nil instead of a function if the whitelist doesn't exist or is empty or something like that.

Reply:
maybe add a comment in the code about why?